ui/app/AppLayouts/Profile/views/LanguageView.qml

```diff
@@ -12,42 +14,127 @@ import "../popups"
 import "../stores"
 
 Item {
-    id: languageContainer
-    Layout.fillHeight: true
-    Layout.fillWidth: true
-    clip: true
+    id: root
 
     property LanguageStore languageStore
     property int profileContentWidth
-    property Component languagePopup: LanguageModal {
-        languageStore: languageContainer.languageStore
+
+    QtObject {
+        id: d
+        property int margins: 64
+        property int zOnTop: 100
+
+        function setViewIdleState() {
+            languagePicker.close()
+        }
     }
 
-    Item {
+    z: d.zOnTop
+    Layout.fillHeight: true
+    Layout.fillWidth: true
+    clip: true
+
+    onVisibleChanged: { if(!visible) d.setViewIdleState()}
+
+    Component.onCompleted: { root.languageStore.initializeLanguageModel() }
+
+    Column {
+        z: d.zOnTop
```

Comment:
is the z order really needed in this case?

Reply:
Yes. Since we are using `StatusListPicker` that contains a **drop-down list**, it is needed to have the drop-down list on top.